docs/ai/conceptual/prompt-engineering-in-dot-net.md

```diff
@@ -0,0 +1,190 @@
+
+Consider this text generation example where prompt is the user input and completion is the model ouput:
+
+Prompt:
+**"The president who served the shortest term was "**
+
+Completion:
+_"Pedro Lascurain."_
+
+Looks right, but what if your app is supposed to help US History students? Pedro Lascurain's 45-minute term is the shortest term for any president, but he served Mexico&mdash;the students are probably looking for _"William Henry Harrsion."_ Clearly, the app could be more helpful to its intended users if you gave it some context.
+
+That's the basic idea of prompt engineering: you add context to the prompt to help the model produce better completions. You can do this by giving the model instructions, examples and cues.
+
+## Instructions
+
+This section explains the use of instructions in prompt engineering.
+
+An instruction is text that tells the model how to respond. An instruction can be a directive (**"You are helping students learn about US history, so talk about the US unless they specifically ask about other countries."**) or an imperative (**"Translate to Tagalog:"**).
+
+Directives are more flexible than imperatives:
+
+- A directive can provide more context than an imperative.
+- You can combine several directives in one instruction.
+- It's usually better to implement a series of steps using a sequence of directives. If you tell the model to output the result of each step, when a problem arises you can easily see which step caused it. Although you can tell the model exactly what steps to follow, you can also just tell the model to break the instruction into steps, and then output the result of each step. This approach is called [chain of thought prompting](chain-of-thought-prompting.md).
+
+Instructions are typically more effective when used with examples. However, when you use both in a prompt you should make sure that the instructions are either above or below the examples for best model performance.
+
+Sometimes GPT models don't follow an instruction the way you expect because it doesn't provide enough context. You can add more context to an instruction by including primary and supporting content. You can include these when you add an instruction, and can add or adjust them after you test your instruction's effect.
+
+The earlier GPT models that generate text or code follow any instructions that you include in a prompt. Newer GPT models that support chat-based apps follow instructions that are in a [system message](https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/advanced-prompt-engineering?pivots=programming-language-chat-completions#system-message) or a user message, but not in an assistant message.
```

Comment:
```suggestion
The earlier GPT models that generate text or code follow any instructions that you include in a prompt. Newer GPT models that support chat-based apps follow instructions that are in a [system message](/azure/ai-services/openai/concepts/advanced-prompt-engineering?pivots=programming-language-chat-completions#system-message) or a user message, but not in an assistant message.
```

Just learned on one of my PRs that you can omit the URL prefix `https://learn.microsoft.com` because these docs link to the learn site by default.

Reply:
I knew that but in prior gigs (not for the same group) the transform fixed those URLs - probably because the writers covered  end-user Word and such. I'll make sure to use relative paths!